blacken code and clean up test-env install, introduce pre-commit

hairmare opened the pull request:
This PR reformats the code using [black](https://black.readthedocs.io/en/stable/), configures isort & flake8 to match and adds [pre-commit](https://pre-commit.com/) to the README as well as to the travis tests. I also took care of fixing or ignoring any issues found by flake8.

My apologies for the huge changeset, all of this does not touch/refactor any of the business logic ~~or logic in setup.py~~ (it touches how requirements are read from txt files in setup.py)... The changes to .travis.yml are kept as minimal as possible.

I plan on taking care of the werkzeug deprecation warnings in a future pull request and this prepares the repo to make it easy to do so. I'll also automate dependency bumping PRs once this is merged.

smlz commented:
Hey Bicku

Finally things calmed down a bit, an I get to review your pull request.

This is really cool! Thanks a lot. Can you rebase the changes once more, with the current version of the project?

Also, I'd like to try black's `--skip-string-normalization` option, as this might significantly reduce the number of changes.

The deprecations are mostly gone by now, as authentication is now done with jwt-tokens instead of session cookies.

smlz commented on .flake8:
```diff
@@ -0,0 +1,33 @@
+[flake8]
```
Put this in `setup.cfg`, to the existing `[flake8]` section

hairmare replied:
done with 9209785b76efe21c56eb3e7df135d09c0028e3ad

smlz commented on .isort.cfg:
```diff
@@ -0,0 +1,7 @@
+[settings]
```
Also put this into `setup.cfg` in a `[isort]` section, to reduce the number of configuration files

hairmare replied:
done

smlz commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,18 @@
+        language: system
+        entry: black
+        types: [python]
+      - id: isort
```
Is there a specific reason run isort after black?

hairmare replied:
isort has more opinions than black, but black is ok with what isort does. Due to isort respecting black formatting it's easier to run it after black since isort has some behaviour that black doesn't like when it encounters blackend code.

smlz commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,18 @@
+      - id: black
+        name: black
+        language: system
+        entry: black
```
Can we add the `--skip-string-normalization` option here?

hairmare replied:
I like the `'` more than the `"` but I hate having to switch to what each project wants even more. If we skip string normalization we'll have a mix of `"` and `'` pretty soon.

smlz commented on requirements-dev.txt:
```diff
@@ -0,0 +1,17 @@
+-r requirements.txt
+black==18.9b0
+codecov==2.0.15
```
Not needed generally

hairmare replied:
pinned to help renovate create explicit PRs to help us notice if changes in our coverage metrics stem from codecov bumps.

smlz commented on .travis.yml:
```diff
@@ -23,8 +23,10 @@ before_install:
   - pip install codecov
 install:
+  - pip install -r requirements-dev.txt
   - pip install  ".[test]" .
```
Can be removed, as we're using the `requirements-dev.txt`

hairmare replied:
removed with 9209785b76efe21c56eb3e7df135d09c0028e3ad

smlz commented on .travis.yml:
```diff
@@ -23,8 +23,10 @@ before_install:
+  - pip install -r requirements-dev.txt
   - pip install  ".[test]" .
 script:
+  - pre-commit run -a
```
With this, the flake8 command on line 34 can be removed

hairmare replied:
done

smlz commented on requirements-dev.txt:
```diff
@@ -0,0 +1,17 @@
+isort==4.3.4
+mock==2.0.0
+pre-commit==1.17.0
+pytest==4.1.1
```
We currently use Python's unittest. Is this a dependency of something?

hairmare replied:
I'm just generally used to running my tests through pytest, it is indeed not currently used.

I might want to use fixtures from pytest at some point and we can mull over those decisions once we are at that point.

smlz commented on setup.py:
```diff
@@ -1,37 +1,35 @@
-        'test': ['tox', 'coverage', 'mock', 'flake8'],
-    },
+    install_requires=["docopt", "mutagen", "six", "Werkzeug"],
+    extras_require={"test": ["tox", "coverage", "mock", "flake8"]},
```
Should match lines in `requirements-dev.txt`

hairmare replied:
Replaced with some black magic (ie reading from reqs in setup.py). Since we do not have plans on large scale distribution of this through pypi this is IMO fine for now.

hairmare commented:
I addressed all the feedbacks except the one about `--skip-string-normalization`. I believe strongly that this needs to be standardized, in fact string normalization is one of the reasons I started this in the first place (after always finding myself using `"` and then changing it to `'` to match with the remaining code). I would have been easier on everyone's eyes if we could have stuck with `'` but this is mostly about delegating such mundane decisions to black, so `"` it is.

spameier commented:
> I addressed all the feedbacks except the one about `--skip-string-normalization`. I believe strongly that this needs to be standardized, in fact string normalization is one of the reasons I started this in the first place (after always finding myself using `"` and then changing it to `'` to match with the remaining code). I would have been easier on everyone's eyes if we could have stuck with `'` but this is mostly about delegating such mundane decisions to black, so `"` it is.

I agree. After all that's why we want to use black.
